File: internal/cmd/output.go
package cmd

import (
	"encoding/json"
	"fmt"
	"io"
	"strings"
)

func printJSON(w io.Writer, value any) error {
	enc := json.NewEncoder(w)
	enc.SetIndent("", "  ")
	return enc.Encode(value)
}

func printText(w io.Writer, text string) error {
	if strings.HasSuffix(text, "\n") {
		_, err := io.WriteString(w, text)
		return err
	}
	_, err := fmt.Fprintln(w, text)
	return err
}

File: src/output.ts
import type { Writable } from "node:stream";

export function printJSON(writer: Writable, value: unknown): void {
  writer.write(`${JSON.stringify(value, null, 2)}\n`);
}

export function printText(writer: Writable, text: string): void {
  if (text.endsWith("\n")) {
    writer.write(text);
    return;
  }
  writer.write(`${text}\n`);
}
